u-boot: Jump to preinit as required by read-only rootfs and use proper filesystem.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Joschka Seydell <joschka@seydell.org>

diff --git a/boot.cmd.in b/boot.cmd.in
--- a/boot.cmd.in
+++ b/boot.cmd.in
@@ -42,13 +42,13 @@
 fi
 
 echo "=== Loading device tree from p${rootfs_partition} to ${fdt_addr} ==="
-ext4load virtio ${virtio_device_num}:${rootfs_partition} ${fdt_addr} /boot/@@KERNEL_DEVICETREE@@
+sqfsload virtio ${virtio_device_num}:${rootfs_partition} ${fdt_addr} /boot/@@KERNEL_DEVICETREE@@
 
 echo "=== Loading kernel image from p${rootfs_partition} to ${kernel_addr_r} ==="
-ext4load virtio ${virtio_device_num}:${rootfs_partition} ${kernel_addr_r} /boot/@@KERNEL_IMAGETYPE@@
+sqfsload virtio ${virtio_device_num}:${rootfs_partition} ${kernel_addr_r} /boot/@@KERNEL_IMAGETYPE@@
 
 echo "=== Setting bootargs (rootfs in p${rootfs_partition} and slot ${rauc_slot}) ==="
-setenv bootargs "${bootargs_default} root=/dev/vda${rootfs_partition} rootfstype=ext4 rauc.slot=${rauc_slot}"
+setenv bootargs "${bootargs_default} root=/dev/vda${rootfs_partition} rootfstype=squashfs init=/sbin/preinit rauc.slot=${rauc_slot}"
 echo "[DEBUG] bootargs: ${bootargs}"
 
 echo "=== Persisting environment in flash ==="
